File: javaFile/insert_at_end.java
class Node {
    int data;
    Node next;
    Node(int data) {
        this.data = data;
        this.next = null;

    }
}

class LinkedList {
    Node head;
    LinkedList() {
        this.head = null;
    }

    void insertAtEnd(int data) {
        Node newNode = new Node(data);
        if (head == null) {
            head = newNode;
            return;
        }
        Node lastNode = head;
        while (lastNode.next != null) {
            lastNode = lastNode.next;

        }
        lastNode.next = newNode;
    }

    void printList() {
        Node temp = head;
        while ( temp != null) {
            System.out.print(temp.data + " ");
            temp = temp.next;
        } 
        System.out.println();
    }
}

public class insert_at_end {
    public static void main(String[] args) {
        LinkedList list = new LinkedList();
        list.insertAtEnd(98);
        list.insertAtEnd(77);
        list.insertAtEnd(45);
        list.insertAtEnd(23);
        list.insertAtEnd(12);
        list.printList();
    }
}

File: stack/stackpy/insertAtEnd.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

class Stack:
    def __init__(self):
        self.head = None
    
    def push(self, data):
        newNode = Node(data)
        if self.head is None:
            self.head = newNode
            return
        current = self.head
        while current.next:
            current = current.next
        current.next = newNode

    def pop(self):
        if self.head is None:
            return
        temp = self.head
        if temp.next is None:
            self.head = None
            del temp
            return
        while temp.next and temp.next.next:
            temp = temp.next
        del temp.next
        temp.next = None
    
    def printStack(self):
        temp = self.head
        while temp:
            print(temp.data, end=" ")
            temp = temp.next
        print()

